### PR TITLE
Remove unnecessary error log messages

### PR DESCRIPTION
**Pull request by kannappanr**
## Description
Remove unnecessary error log messages as part of logger improvement.
## Motivation and Context
Corresponding errors are passed to the client and the server is behaving correctly, so the errors need not be logged on the console.

## How Has This Been Tested?
Manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by harshavardhana — DISMISSED** [review]

**Comment by kannappanr** [comment]
@harshavardhana resolved a conflict. PTAL.
@vadmeste Can you please review?

File: cmd/fs-v1-helpers.go [hunk 1]
```diff
@@ -345,7 +345,6 @@ func fsCreateFile(ctx context.Context, filePath string, reader io.Reader, buf []
 	if buf != nil {
 		bytesWritten, err = io.CopyBuffer(writer, reader, buf)
 		if err != nil {
-			logger.LogIf(ctx, err)
```
Comment (vadmeste) [inline cmd/fs-v1-helpers.go:348]: This is creating a new file and directly write on the disk, I think it makes sense to keep this logged, I don't know any error that we need to ignore here. We can at least capture i/o error here.

File: cmd/fs-v1.go [hunk 2]
```diff
@@ -672,7 +672,6 @@ func (fs *FSObjects) getObjectInfo(ctx context.Context, bucket, object string) (
 
 	// Ignore if `fs.json` is not available, this is true for pre-existing data.
 	if err != nil && err != errFileNotFound {
-		logger.LogIf(ctx, err)
```
Comment (vadmeste) [inline cmd/fs-v1.go]: I am not why we removed this one, what boring error are we currently seeing? I want to keep this to be able to see some i/o errors or permission error

File: cmd/fs-v1.go [hunk 3]
```diff
@@ -828,7 +827,6 @@ func (fs *FSObjects) putObject(ctx context.Context, bucket string, object string
 		fsMetaPath := pathJoin(bucketMetaDir, bucket, object, fs.metaJSONFile)
 		wlk, err = fs.rwPool.Create(fsMetaPath)
 		if err != nil {
-			logger.LogIf(ctx, err)
```
Comment (vadmeste) [inline cmd/fs-v1.go]: With my basic testing, I didn't see this is logged at all. Is this something we need to really ignore ?

File: cmd/xl-v1-utils.go [hunk 4]
```diff
@@ -297,9 +293,6 @@ func readXLMeta(ctx context.Context, disk StorageAPI, bucket string, object stri
 	// Reads entire `xl.json`.
 	xlMetaBuf, err := disk.ReadAll(bucket, path.Join(object, xlMetaJSONFile))
 	if err != nil {
-		if err != errFileNotFound {
-			logger.LogIf(ctx, err)
```
Comment (vadmeste) [inline cmd/xl-v1-utils.go]: Can't we log this like, once in a while ? it is good to see problems related to access to other nodes

**Review by vadmeste — COMMENTED** [review]
Some comments

File: cmd/fs-v1-helpers.go (hunk 1 above)
Comment (kannappanr) [reply cmd/fs-v1-helpers.go:348]: This log was generated by running `mint` test in `FS` mode.
```
API: PutObjectPart(bucket=minio-java-test-ti02s0, object=minio-java-test-33240mb)
Time: 16:25:11 PDT 07/12/2018
RemoteHost: 172.17.0.2
UserAgent: Minio (amd64; amd64) minio-java/dev
Error: unexpected EOF
       1: cmd/fs-v1-helpers.go:346:cmd.fsCreateFile()
       2: cmd/fs-v1-multipart.go:319:cmd.(*FSObjects).PutObjectPart()
       3: cmd/object-handlers.go:1314:cmd.(ObjectLayer).PutObjectPart-fm()
       4: cmd/object-handlers.go:1318:cmd.objectAPIHandlers.PutObjectPartHandler()
       5: cmd/api-router.go:63:cmd.(objectAPIHandlers).PutObjectPartHandler-fm()
       6: net/http/server.go:1947:http.HandlerFunc.ServeHTTP()

```

The tests passed, but we were seeing this error on the server side, which is not correct.

File: cmd/fs-v1.go (hunk 2 above)
Comment (kannappanr) [reply cmd/fs-v1.go]: We can maybe keep this one. If we see that this is logging quite a bit, we can remove it.

File: cmd/fs-v1.go (hunk 3 above)
Comment (kannappanr) [reply cmd/fs-v1.go]: Same comment as above. Can keep it for now, and make a decision on this later.

File: cmd/xl-v1-utils.go (hunk 4 above)
Comment (kannappanr) [reply cmd/xl-v1-utils.go]: I can change this to call `LogOnceIf`

File: cmd/fs-v1-helpers.go (hunk 1 above)
Comment (kannappanr) [reply cmd/fs-v1-helpers.go:348]: For now, I am not printing only on `io.ErrUnexpectedEOF`

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Review by harshavardhana — APPROVED** [review]

**Review by poornas — APPROVED** [review]

**Pull request merged**